cmd/kubeadm/app/util/endpoint.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+)
+
+// GenerateMasterEndpoint generates a properly formatted Master URL.
+func GenerateMasterEndpoint(address string, port int32) string {
+	var masterEndpoint string
```

Comment:
Use the existing net.JoinHostPort() method to create the Endpoint string, instead of the logic below. You just need to convert the port to a string (I hit the same comment in one of my commits).

Ideally, the caller could check that the "address" is valid, and this could have the sole purpose of creating the URL. If adding check here (for nil), then consider having this function return (string, error). But really, it is cleaner IMHO, to have this just generate the URL and the validation (which is just a simple check that address is valid IP) to happen elsewhere.